mumax3c/scripts/energy.py

```diff
@@ -40,27 +41,40 @@ def zeeman_script(system):
 
 
 # Needs to be tidied up
-def uniaxialanisotropy_script(system):
+def uniaxialanisotropy_script(system, name):
     mx3 = "// UniaxialAnisotropy\n"
-    mx3 += f"Ku1 = {system.energy.uniaxialanisotropy.K1}\n"
-    # K2 to be added
-    mx3 += "anisu = vector({}, {}, {})\n\n".format(*system.energy.uniaxialanisotropy.u)
+    if hasattr(getattr(system.energy, name), "K"):
+        mx3 += mc.scripts.set_parameter(
+            parameter=getattr(system.energy, name).K, name="Ku1", system=system
+        )
+    else:
+        mx3 += mc.scripts.set_parameter(
+            parameter=getattr(system.energy, name).K1, name="Ku1", system=system
+        )
+        mx3 += mc.scripts.set_parameter(
+            parameter=getattr(system.energy, name).K2, name="Ku2", system=system
+        )
 
+    mx3 += mc.scripts.set_parameter(
+        parameter=getattr(system.energy, name).u, name="anisU", system=system
+    )
     return mx3
 
 
-def demag_script(system):
+def demag_script(system, name):
     mx3 = "// Demag\n"
     mx3 += "enabledemag = true\n\n"
     return mx3
 
 
-def dmi_script(system):
+def dmi_script(system, name):
     mx3 = ""
     if system.energy.dmi.crystalclass.lower() in ["t", "o"]:
-        name = "Dbulk"
-    elif system.energy.dmi.crystalclass.lower() == "cnv":
-        name = "Dind"
+        param_name = "Dbulk"
+        param_val = getattr(system.energy, name).D
+    elif system.energy.dmi.crystalclass.lower() in ["cnv_z", "cnv"]:
+        param_name = "Dind"
+        param_val = -getattr(system.energy, name).D
```

Comment:
`-D` does not work if `D` is a dictionary.

Reply:
Thanks for pointing this out!